Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -322,6 +322,21 @@ TEST_F(SerializerTest, BadStringValue) {
       Status(FirestoreErrorCode::InvalidArgument, "ignored"), bytes);
 }
 
+TEST_F(SerializerTest, BadStringValue2) {
+  std::vector<uint8_t> bytes;
```

Comment:
I really want to make the suggestion that this bit can be a helper, but I think the typing chrome ends up making this worse:

```c++
std::vector<uint8_t> ExpectSuccess(std::function<Status(std::vector<uint8_t>*)> encoder) {
  std::vector<uint8_t> bytes;
  Status status = encoder(&bytes);
  ASSERT_TRUE(status.ok);
  return bytes;
}

TEST_F(SerializerTest, BadStringValue3) {
  std::vector<uint8_t> bytes = ExpectSuccess([](std::vector<uint8_t>* bytes) -> Status {
    return serializer.EncodeFieldValue(FieldValue::StringValue("a"), bytes);
  });

  // ...
}
```
It could be made better with a typedef of `std::vector<uint8_t>` to `Bytes` (this could probably be done even in the models), but what you have here is likely as good as it gets. I've included this just as an FYI; maybe you can make something of it.

Reply:
Good idea, but it doesn't quite work. This:

```c++
T fn() {
  [] { ASSERT_TRUE(false); }();
  EXPECT_TRUE(false);
}
```

reports two errors. So it looks like the 'assert' effectively gets turned into an 'expect' in this situation.

Reply:
Sorry, that was a bad suggestion on my part.

The difference between ASSERT and EXPECT is that ASSERT returns from the current function (this is why it requires a void return type). You can't really use ASSERT and EXPECT within helper methods to force termination (my bad).

Re: ExpectSuccessfullyEncodedFieldValue, LGTM except for the name, which is absurdly long:
  * It's already a locally defined test helper so we could probably drop the "Expect"
  * s/SuccessfullyEncoded/Encode/ ?

Basically: this is something we don't expect to fail. We happen to be verifying it didn't using the test machinery to do so, but this isn't something a test author would reuse to actually test anything. Therefore I think the name is overkill.

I suggest just `EncodedFieldValue`. It's shorter, doesn't suggest that it should be used for verifying the SUT, but still conveys enough about what it's doing. The type of the parameters and the fact that it's defined in a test as a helper do enough to discourage misuse.